roles/openshift_clock_facts/tasks/main.yaml

```diff
@@ -0,0 +1,40 @@
+---
+- fail:
+    msg: "openshift_clock_method must be set to either 'none', 'auto', 'ntpd', or 'chrony'"
+  when:  openshift_clock_method is defined and
+    openshift_clock_method not in ["none", "auto", "ntpd", "chrony"]
+
+- set_fact:
+    openshift_clock_method: chrony
+  when: openshift_clock_method is defined and openshift_clock_method == "auto" and
+    openshift.common.is_containerized and openshift.common.is_atomic
+
+- set_fact:
+    openshift_clock_method: ntpd
+  when: openshift_clock_method is defined and openshift_clock_method == "auto"
```

Comment:
You could also push the logic for determining whether 'auto' should be 'ntpd' or 'chrony' into openshift_facts.